edk2toollib/windows/telem/cper_parser_tool.py

```diff
@@ -0,0 +1,1001 @@
+                    "9a78788a-bbe8-11e4-809e-67611e5d46b0": "SEA Notify Type",
+                    "5c284c81-b0ae-4e87-a322-b04c85624323": "SEI Notify Type",
+                    "09a9d5ac-5204-4214-96e5-94992e752bcd": "PEI Notify Type",
+                    "487565ba-6494-4367-95ca-4eff893522f6": "BMC Notify Type",
+                    "e9d59197-94ee-4a4f-8ad8-9b7d8bd93d2e": "SCI Notify Type",
+                    "fe84086e-b557-43cf-ac1b-17982e078470": "EXTINT Notify Type",
+                    "0033f803-2e70-4e88-992c-6f26daf3db7a": "Device Driver Notify Type",
+                    "919448b2-3739-4b7f-a8f1-e0062805c2a3": "CMCI Notify Type",
+                    "9876ccad-47b4-4bdb-b65e-16f193c4f3db": "Processor Generic Error Section",
+                    "00000000-0000-0000-0000-000000000000": "Device Driver Error Section",
+                    "1c15b445-9b06-4667-ac25-33c056b88803": "IPMI MSR Dump Section",
+                    "dc3ea0b0-a144-4797-b95b-53fa242b6e1d": "XPF Processor Error",
+                    "e429faf1-3cb7-11d4-bca7-0080c73c8881": "IPF Processor Error",
+                    "e19e3d16-bc11-11e4-9caa-c2051d5d46b0": "ARM Processor Error",
+                    "a5bc1114-6f64-4ede-b863-3e83ed7c83b1": "Memory Error Section",
+                    "d995e954-bbc1-430f-ad91-b44dcb3c6f35": "PCIe Error Section",
+                    "c5753963-3b84-4095-bf78-eddad3f9c9dd": "PCIX Bus Error Section",
+                    "eb5e4685-ca66-4769-b6a2-26068b001326": "PCIX Device Error Section",
+                    "81212a96-09ed-4996-9471-8d729c8e69ed": "Firmware Error Record Reference",
+                    "81687003-dbfd-4728-9ffd-f0904f97597d": "PMEM Error Section",
+                    "a55701f5-e3ef-43de-ac72-249b573fad2c": "WHEA Cache Check",
+                    "fc06b535-5e1f-4562-9f25-0a3b9adb63c3": "WHEA TLB Check",
+                    "1cf3f8b3-c5b1-49a2-aa59-5eef92ffa63c": "WHEA Bus Check",
+                    "48ab7f57-dc34-4f6c-a7d3-b0b5b0a74314": "WHEA MS Check",
+                    "cf07c4bd-b789-4e18-b3c4-1f732cb57131": "WHEA Record Creator",
+                    "3e62a467-ab40-409a-a698-f362d464b38f": "Generic Notfiy Type",
+                    "6f3380d1-6eb0-497f-a578-4d4c65a71617": "IPF SAL Record Section",
+                    "8a1e1d01-42f9-4557-9c33-565e5cc3f7e8": "XPF MCA Section",
+                    "e71254e7-c1b9-4940-ab76-909703a4320f": "NMI Section",
+                    "e71254e8-c1b9-4940-ab76-909703a4320f": "Generic Section",
+                    "e71254e9-c1b9-4940-ab76-909703a4320f": "WHEA Error Packet Section",
+                    }
+
+# List of plugin classes which inherit from CPER_SECTION_DATA and are
+# therefore capable of parsing section data
+Parsers = SECTION_DATA_PARSER.__subclasses__()
+
+
+class GENERIC_FIELD(object):
+    '''A super class for fields of CPER records'''
+
+    def __init__(self, raw_value):
+        self.raw_value = raw_value
```

Comment:
This pattern doesn't exist anywhere else in the encoding object we have in edk2toollib (to my knowledge). The object themselves should just be shells that contain the accessible fields. Then you should call `decode()` on either an `IOStream` or a `bytes` object (Sean and I disagree on this, but either approach is fine) to populate the contents. Then, ideally, there should also be an `encode()` method that would put the object back into either `bytes` or an `IOStream` such that `MyObject.decode(my_bytes).encode() == my_bytes`. But just having the `decode()` method is fine for a parser.